src/lib/comms/include/sol-network.h

```diff
@@ -234,31 +235,66 @@ static inline bool
 sol_network_link_addr_eq(const struct sol_network_link_addr *a,
     const struct sol_network_link_addr *b)
 {
-    const uint8_t *addr_a, *addr_b;
     size_t bytes;
 
-    if (a->family != b->family)
-        return false;
+    if (a->family == b->family) {
+        const uint8_t *addr_a, *addr_b;
+
+        if (a->family == SOL_NETWORK_FAMILY_INET) {
+            addr_a = a->addr.in;
+            addr_b = b->addr.in;
+            bytes = sizeof(a->addr.in);
+        } else if (a->family == SOL_NETWORK_FAMILY_INET6) {
+            addr_a = a->addr.in6;
+            addr_b = b->addr.in6;
+            bytes = sizeof(a->addr.in6);
+        } else if (a->family == SOL_NETWORK_FAMILY_BLUETOOTH) {
+            if (a->addr.bt_type != b->addr.bt_type)
+                return false;
+
+            addr_a = a->addr.bt_addr;
+            addr_b = b->addr.bt_addr;
+            bytes = sizeof(a->addr.bt_addr);
+        } else
+            return false;
+        return !memcmp(addr_a, addr_b, bytes);
+    }
+
+    if ((a->family == SOL_NETWORK_FAMILY_INET &&
+         b->family == SOL_NETWORK_FAMILY_INET6) ||
+        (a->family == SOL_NETWORK_FAMILY_INET6 &&
+         b->family == SOL_NETWORK_FAMILY_INET)) {
+
+        struct ipv6_map_prefix {
+            const uint8_t zeroes[10];
+            const uint16_t ones;
```

Comment:
From `netinet/in.h`:

```C
# define IN6_IS_ADDR_V4MAPPED(a) \
  (__extension__							      \
   ({ const struct in6_addr *__a = (const struct in6_addr *) (a);	      \
      __a->s6_addr32[0] == 0						      \
      && __a->s6_addr32[1] == 0						      \
      && __a->s6_addr32[2] == htonl (0xffff); }))
```

The use of `htonl()` and the `s6_addr32` name makes me think that this should be 32-bit wide.